### PR TITLE
openharmony bug fix

### PR DESCRIPTION
**Pull request by zhaoyang57**
1.fix custom font register timing. 2.fix finalize bug for openharmony napi wrapper.

### CODE REVIEW

File: cocos/scripting/js-bindings/manual/jsb_platform.h [hunk 1]
```diff
@@ -25,11 +25,16 @@
 
 #pragma once
 
+#include <native_drawing/drawing_text_declaration.h>
```
Comment (qiuguohua) [inline cocos/scripting/js-bindings/manual/jsb_platform.h:29]: Here we need to add the openharmony macro？

File: cocos/platform/openharmony/modules/CCCanvasRenderingContext2D-openharmony.cpp [hunk 2]
```diff
@@ -245,13 +227,25 @@ class CanvasRenderingContext2DImpl {
...
-            fontFamilies[0] = fontName.c_str();
-            OH_Drawing_SetTextStyleFontFamilies(_textStyle, 1, fontFamilies);
-            OH_Drawing_SetTextStyleLocale(_textStyle, "en");
+        
+        const auto &fontInfoMap = getFontFamilyCollectionMap();
+        auto it = fontInfoMap.find(_fontName);
+        if (it != fontInfoMap.end()) {
+            _fontCollection = it->second;
+        } else {
+            _fontCollection = OH_Drawing_CreateSharedFontCollection();
```
Comment (qiuguohua) [inline cocos/platform/openharmony/modules/CCCanvasRenderingContext2D-openharmony.cpp:246]: If it run to this condition, what is the font displayed by that fontCollection?

File: cocos/scripting/js-bindings/manual/jsb_platform.h (hunk 1 above)
Comment (zhaoyang57) [reply cocos/scripting/js-bindings/manual/jsb_platform.h:29]: fixed.

File: cocos/platform/openharmony/modules/CCCanvasRenderingContext2D-openharmony.cpp (hunk 2 above)
Comment (zhaoyang57) [reply cocos/platform/openharmony/modules/CCCanvasRenderingContext2D-openharmony.cpp:246]: will use system font

Comment (qiuguohua) [reply cocos/platform/openharmony/modules/CCCanvasRenderingContext2D-openharmony.cpp:246]: Got it.

File: cocos/platform/openharmony/modules/CCCanvasRenderingContext2D-openharmony.cpp [hunk 3]
```diff
@@ -245,13 +227,25 @@ class CanvasRenderingContext2DImpl {
...
         }
+        _typographyStyle = OH_Drawing_CreateTypographyStyle();
```
Comment (qiuguohua) [inline cocos/platform/openharmony/modules/CCCanvasRenderingContext2D-openharmony.cpp:248]: Doesn't _textStyle need to be released first if it's repeated with different fonts?

Comment (zhaoyang57) [reply cocos/platform/openharmony/modules/CCCanvasRenderingContext2D-openharmony.cpp:248]: You are right, fixed already.

File: cocos/platform/openharmony/modules/CCCanvasRenderingContext2D-openharmony.cpp [hunk 4]
```diff
@@ -245,13 +227,35 @@ class CanvasRenderingContext2DImpl {
...
-            OH_Drawing_SetTextStyleFontFamilies(_textStyle, 1, fontFamilies);
-            OH_Drawing_SetTextStyleLocale(_textStyle, "en");
+        
+        if(_fontCollection) {
+            OH_Drawing_DestroyFontCollection(_fontCollection);
+        }
+        if(_typographyCreate) {
+            OH_Drawing_DestroyTypographyHandler(_typographyCreate);
+        }
+        if(_textStyle) {
```
Comment (qiuguohua) [inline cocos/platform/openharmony/modules/CCCanvasRenderingContext2D-openharmony.cpp]: Does _typographyCreate need to be released?

File: cocos/platform/openharmony/modules/CCCanvasRenderingContext2D-openharmony.cpp [hunk 5]
```diff
@@ -245,13 +227,35 @@ class CanvasRenderingContext2DImpl {
...
-        if (!_fontName.empty()) {
-            const char* fontFamilies[1];
-            fontFamilies[0] = fontName.c_str();
-            OH_Drawing_SetTextStyleFontFamilies(_textStyle, 1, fontFamilies);
-            OH_Drawing_SetTextStyleLocale(_textStyle, "en");
+        
+        if(_fontCollection) {
+            OH_Drawing_DestroyFontCollection(_fontCollection);
+        }
+        if(_typographyCreate) {
```
Comment (qiuguohua) [inline cocos/platform/openharmony/modules/CCCanvasRenderingContext2D-openharmony.cpp]:
```suggestion
        if (_typographyCreate) {
```

File: cocos/platform/openharmony/modules/CCCanvasRenderingContext2D-openharmony.cpp (hunk 4 above)
Comment (zhaoyang57) [reply cocos/platform/openharmony/modules/CCCanvasRenderingContext2D-openharmony.cpp]: The creation of _typographyCreate depends on _fontCollection parameter, so need to be released together.

**Review by qiuguohua — APPROVED** [review]

File: cocos/platform/openharmony/modules/CCCanvasRenderingContext2D-openharmony.cpp [hunk 6]
```diff
@@ -245,13 +227,35 @@ class CanvasRenderingContext2DImpl {
...
+        _typographyStyle = OH_Drawing_CreateTypographyStyle();
+        OH_Drawing_SetTypographyTextDirection(_typographyStyle, TEXT_DIRECTION_LTR);
+        OH_Drawing_SetTypographyTextAlign(_typographyStyle, TEXT_ALIGN_LEFT);
+        _typographyCreate = OH_Drawing_CreateTypographyHandler(_typographyStyle, _fontCollection);
+        _textStyle = OH_Drawing_CreateTextStyle();
+
+        const char *fontFamilies[1];
+        fontFamilies[0] = fontName.c_str();
+        OH_Drawing_SetTextStyleFontFamilies(_textStyle, 1, fontFamilies);
+        OH_Drawing_SetTextStyleLocale(_textStyle, "en");
```
Comment (minggo) [inline cocos/platform/openharmony/modules/CCCanvasRenderingContext2D-openharmony.cpp]: Only support english?

Comment (zhaoyang57) [reply cocos/platform/openharmony/modules/CCCanvasRenderingContext2D-openharmony.cpp]: There is no need to call this interface. Removed already.

**Review by minggo — APPROVED** [review]

**Pull request merged**